# Comparative Analysis of Descriptive Metadata

### Newcastle University Special Collections and University of Edinburgh Archives

***

**Table of Contents**
* [Description Lengths](#description-lengths)
* [Gendered Language](#gendered-language)
* [Parts of Speech](#parts-of-speech)
  * [Adjectives](#adjectives)
  * [Adverbs](#adverbs)

***

In [108]:
import analysis_utils
import pandas as pd
import numpy as np
from pathlib import Path

Create variables to reference the files and data points from the `analysis_metadata_XXX.ipynb` notebooks.

In [39]:
dir_nusc = "data/analysis/"
f_descs_nusc = dir_nusc+"nusc_ead_descs_tokenized.csv"
files_nusc = [
    "nusc_ead_descs_stats.csv", 
    "nusc_ead_gendered_lower_token_counts_ead.csv", "nusc_ead_gendered_capitalized_token_counts.csv",
    "nusc_ead_descs_pos_tags.csv",
    "nusc_ead_nltk_adj_token_counts.csv", "nusc_ead_nltk_adj_token_lower_counts.csv",
    "nusc_ead_nltk_adv_token_counts.csv", "nusc_ead_nltk_adv_token_lower_counts.csv",
    "nusc_ead_nltk_adj_adv_counts.csv", "nusc_ead_nltk_adj_adv_stats.csv"
]

In [40]:
dir_uoe = "data/uoe/analysis/"
f_descs_uoe = "data/uoe/analysis/uoe_descs.csv"
files_uoe = [
    "uoe_ead_descs-oct2020_stats.csv", 
    "uoe_ead_gendered_lower_token_counts_oct2020.csv", "uoe_ead_gendered_capitalized_token_counts_oct2020.csv",
    "uoe_ead_descs_oct2020_pos_tags.csv", 
    "uoe_descs_oct2020_nltk_adj_token_counts.csv", "uoe_descs_oct2020_nltk_adj_token_lower_counts.csv",
    "uoe_descs_oct2020_nltk_adv_token_counts.csv", "uoe_descs_oct2020_nltk_adv_token_lower_counts.csv",
    "uoe_descs_oct2020_nltk_adj_adv_counts.csv", "uoe_oct2020_nltk_adj_adv_stats.csv"
]

In [4]:
pd.set_option('display.float_format', lambda x: '%.3f' % x)  # display floating point numbers with 3 decimal places

In [109]:
comparison_dir = "data/nusc_vs_uoe/"
Path(comparison_dir).mkdir(parents=True, exist_ok=True)

## Description Lengths

In [13]:
df_len_nusc = pd.read_csv(dir_nusc + "nusc_ead_descs_stats.csv", index_col=0)
df_len_nusc

sentences_per_description  tokens_per_description  \
mean                                    1.658                  25.482   
median                                  1.000                   9.000   
standard_deviation                      2.396                  63.481   
minimum                                 1.000                   0.000   
maximum                                91.000                2296.000   
total                              124592.000             1915370.000   

                    tokens_per_sentence  
mean                             15.373  
median                           12.000  
standard_deviation               14.191  
minimum                           0.000  
maximum                        1563.000  
total                       1915370.000

In [14]:
df_len_uoe = pd.read_csv(dir_uoe + "uoe_ead_descs_oct2020_stats.csv", index_col=0)
df_len_uoe

sentences_per_description  tokens_per_description  \
mean                                    1.502                  19.553   
median                                  1.000                  10.000   
standard_deviation                      5.377                  97.627   
minimum                                 1.000                   1.000   
maximum                               742.000               14097.000   
total                               41671.000              542635.000   

                    tokens_per_sentence  
mean                             13.022  
median                            9.000  
standard_deviation               14.975  
minimum                           1.000  
maximum                         551.000  
total                        542635.000

Newcastle's data shows more tokens per description on average (25.482 vs. 19.553), which I hadn't expected, though the median is greater for Edinburgh (10 vs. 9).  There is also greater variation in description lengths in the Edinburgh data relative to the Newcastle data (for words per description, the standard deviation of Edinburgh's is 97.627 vs. Newcastle's at 63.481; for sentences per description, Edinburgh's is 5.377 vs. Newcastle's of 2.396).  Also, the maximum number of tokens in a single description is far larger (14,097 vs. 2,296 - over 6x as large), as is the maximum number of sentences (742 vs. 91).  The average tokens per sentence are similar (15.373 for Edinburgh vs. 13.022 for Newcastle).

What if we remove all descriptions that aren't more than 1 sentence long.  How many descriptions are we left with in the Newcastle and Edinburgh datasets?  What is the average length of the remaining descriptions?

In [35]:
df_descs_nusc = pd.read_csv(f_descs_nusc, index_col=0)
# The tokenized column will be read as a string, so convert it to a list of lists of strings
df_descs_nusc = analysis_utils.getColumnValuesAs2dList(df_descs_nusc, "tokenized", replace=True)
df_descs_nusc.head()

description_id         eadid         rowid         field  \
0               0  17th C. Coll  17th C. Coll      bioghist   
1               1  17th C. Coll  17th C. Coll  scopecontent   
2               2  17th C. Coll  17th C. Coll     unittitle   
3               3  18th C. Coll  18th C. Coll      bioghist   
4               4  18th C. Coll  18th C. Coll  scopecontent   

                                                 doc  \
0  Formed in 1963 after an amalgamation of instit...   
1  The 17th Century Collection is a small but exp...   
2                            17th Century Collection   
3  Formed in 1963 after an amalgamation of instit...   
4  The 18th Century Collection contains approxima...   

                                           tokenized  
0  [[Formed, in, 1963, after, an, amalgamation, o...  
1  [[The, 17th, Century, Collection, is, a, small...  
2                      [[17th, Century, Collection]]  
3  [[Formed, in, 1963, after, an, amalgamation, o...  
4  [[The, 18th, Century, Collection, contains, ap...

In [36]:
tokenized_nusc = list(df_descs_nusc["tokenized"])
sents_per_desc_nusc = [len(desc) for desc in tokenized_nusc]
# tokens_per_desc_nusc = [sum([len(sent) for sent in desc]) for desc in tokenized_nusc]
print(sents_per_desc_nusc[0:5])
# print(tokens_per_desc_nusc[0:5])

[2, 3, 1, 2, 3]


In [37]:
assert len(sents_per_desc_nusc) == df_descs_nusc.shape[0], "There should be one sentence count for every description."

In [38]:
df_descs_uoe = pd.read_csv(f_descs_uoe, index_col=0)
df_descs_uoe = analysis_utils.getColumnValuesAsLists(df_descs_uoe, "sentence_id")
df_descs_uoe.head()

description_id                sentence_id  \
0               0                        [0]   
1               1                        [1]   
2               2                        [2]   
3               3  [3, 4, 5, 6, 7, 8, 9, 10]   
4               4                       [11]   

                                            token_id  \
0                                              [[2]]   
1                  [[5, 6, 7, 8, 9, 10, 11, 12, 14]]   
2  [[17, 20, 21, 22, 24, 26, 28, 30, 31, 32, 33, ...   
3  [[113, 114, 115, 116, 117, 118, 119, 120, 121,...   
4                                            [[310]]   

                                               token  \
0                                          [['AA5']]   
1  [['Papers', 'of', 'The', 'Very', 'Rev', 'Prof'...   
2  [['and', 'Sermons', 'and', 'addresses', '1948-...   
3  [['Professor', 'James', 'Aitken', 'White', 'wa...   
4                                          [['AA6']]   

                                                 pos  
0                                           [['NN']]  
1  [['NNS', 'IN', 'DT', 'NNP', 'NNP', 'NNP', 'NNP...  
2  [['CC', 'NNS', 'CC', 'NNS', 'JJ', 'NNS', 'JJ',...  
3  [['NNP', 'NNP', 'NNP', 'NNP', 'VBD', 'DT', 'JJ...  
4                                           [['NN']]

In [39]:
sent_id_lists = list(df_descs_uoe.sentence_id)
sents_per_desc_uoe = [len(sent_ids) for sent_ids in sent_id_lists]
print(sents_per_desc_uoe[0:5])

[1, 1, 1, 8, 1]


In [40]:
long_sents_nusc = [s_count for s_count in sents_per_desc_nusc if s_count > 1]
print("Remaining NUSC descriptions:", len(long_sents_nusc), "or", str((len(long_sents_nusc)/len(sents_per_desc_nusc))*100)+"% of total")
long_sents_uoe = [s_count for s_count in sents_per_desc_uoe if s_count > 1]
print("Remaining UoE descriptions:", len(long_sents_uoe), "or", str(len(long_sents_uoe)/len(sents_per_desc_uoe)*100)+"% of total")

Remaining NUSC descriptions: 15422 or 20.51752810483603% of total
Remaining UoE descriptions: 5554 or 20.01297203805131% of total


About 20% of **both** dataset's descriptions are 2 or more sentences long!

In [41]:
long_sents_nusc = [s_count for s_count in sents_per_desc_nusc if s_count > 2]
print("Remaining NUSC descriptions:", len(long_sents_nusc), "or", str((len(long_sents_nusc)/len(sents_per_desc_nusc))*100)+"% of total")
long_sents_uoe = [s_count for s_count in sents_per_desc_uoe if s_count > 2]
print("Remaining UoE descriptions:", len(long_sents_uoe), "or", str(len(long_sents_uoe)/len(sents_per_desc_uoe)*100)+"% of total")

Remaining NUSC descriptions: 6250 or 8.315040244794785% of total
Remaining UoE descriptions: 1899 or 6.8427500720668775% of total


In [42]:
long_sents_nusc = [s_count for s_count in sents_per_desc_nusc if s_count > 3]
print("Remaining NUSC descriptions:", len(long_sents_nusc), "or", str((len(long_sents_nusc)/len(sents_per_desc_nusc))*100)+"% of total")
long_sents_uoe = [s_count for s_count in sents_per_desc_uoe if s_count > 3]
print("Remaining UoE descriptions:", len(long_sents_uoe), "or", str(len(long_sents_uoe)/len(sents_per_desc_uoe)*100)+"% of total")

Remaining NUSC descriptions: 4132 or 5.497239406638728% of total
Remaining UoE descriptions: 982 or 3.538483712885558% of total


## COME BACK TO: B/H only!

## Gendered Language

In [23]:
i = 1 # lowercased grammatically/lexically gendered tokens

In [24]:
df_nusc_gendered = pd.read_csv(dir_nusc + files_nusc[i], index_col=0)
print(df_nusc_gendered.shape)

(30, 1)


In [25]:
df_uoe_gendered = pd.read_csv(dir_uoe + files_uoe[i], index_col=0)
print(df_uoe_gendered.shape)

(30, 1)


In [26]:
total_tokens_nusc = 1915370
perc_nusc = (df_nusc_gendered["count"]/total_tokens_nusc)*100
df_nusc_gendered.insert(len(df_nusc_gendered.columns), "percentage", perc_nusc)
df_nusc_gendered

count  percentage
word                            
him                0       0.000
his                0       0.000
hers               0       0.000
her                0       0.000
she                0       0.000
he                 0       0.000
granddaughter      3       0.000
niece              3       0.000
grandmother        6       0.000
grandson           7       0.000
nephew             8       0.000
grandfather       16       0.001
boys              33       0.002
uncle             34       0.002
aunt              44       0.002
girls             51       0.003
husband           57       0.003
girl              70       0.004
boy               75       0.004
brother           93       0.005
daughter         100       0.005
sister           180       0.009
son              183       0.010
wife             196       0.010
woman            198       0.010
women            345       0.018
mother           415       0.022
man              429       0.022
father           488       0.025
men              975       0.051

In [27]:
df_nusc_gendered.describe()

count  percentage
count  30.000      30.000
mean  133.633       0.007
std   212.657       0.011
min     0.000       0.000
25%     3.750       0.000
50%    47.500       0.002
75%   182.250       0.010
max   975.000       0.051

In [28]:
total_tokens_uoe = 542635
perc_uoe = (df_uoe_gendered["count"]/total_tokens_uoe)*100
df_uoe_gendered.insert(len(df_uoe_gendered.columns), "percentage", perc_uoe)
df_uoe_gendered

count  percentage
word                            
he                 0       0.000
niece              0       0.000
his                0       0.000
hers               0       0.000
granddaughter      0       0.000
him                0       0.000
her                0       0.000
she                0       0.000
aunt               2       0.000
grandmother        4       0.001
grandson           4       0.001
nephew             5       0.001
grandfather        8       0.001
uncle             15       0.003
girls             23       0.004
husband           23       0.004
sister            27       0.005
girl              32       0.006
boys              48       0.009
boy               52       0.010
mother            53       0.010
daughter          60       0.011
brother           65       0.012
father           106       0.020
wife             109       0.020
son              145       0.027
women            157       0.029
woman            279       0.051
men              452       0.083
man              656       0.121

In [29]:
df_uoe_gendered.describe()

count  percentage
count  30.000      30.000
mean   77.500       0.014
std   146.203       0.027
min     0.000       0.000
25%     0.500       0.000
50%    23.000       0.004
75%    63.750       0.012
max   656.000       0.121

The use of grammatically and lexically gendered terms is a small percentage of all tokens in both datasets, but Edinburgh's descriptions relative to Newcastle's are larger.

In [30]:
i = 2 # capitalized grammatically/lexically gendered tokens

In [31]:
df_nusc_gendered = pd.read_csv(dir_nusc + files_nusc[i], index_col=0)
print(df_nusc_gendered.shape)

(25, 1)


In [32]:
df_uoe_gendered = pd.read_csv(dir_uoe + files_uoe[i], index_col=0)
print(df_uoe_gendered.shape)

(25, 1)


In [33]:
total_tokens_nusc = 1915370
perc_nusc = (df_nusc_gendered["count"]/total_tokens_nusc)*100
df_nusc_gendered.insert(len(df_nusc_gendered.columns), "percentage", perc_nusc)
df_nusc_gendered

count  percentage
word                          
Viscountess      1       0.000
gentlemen        2       0.000
gentleman        4       0.000
ladies           5       0.000
Duchess          5       0.000
Ms               5       0.000
Baroness         6       0.000
Countess         6       0.000
Count            8       0.000
Viscount        11       0.001
lady            13       0.001
Princess        16       0.001
Messrs          17       0.001
Baron           22       0.001
Duke            28       0.001
Dame            34       0.002
Prince          53       0.003
Miss           115       0.006
Mrs            127       0.007
Queen          142       0.007
King           162       0.008
Mr             204       0.011
Lord           246       0.013
Lady           579       0.030
Sir           2161       0.113

In [34]:
df_nusc_gendered.describe()

count  percentage
count   25.000      25.000
mean   158.880       0.008
std    435.808       0.023
min      1.000       0.000
25%      6.000       0.000
50%     17.000       0.001
75%    127.000       0.007
max   2161.000       0.113

In [35]:
total_tokens_uoe = 542635
perc_uoe = (df_uoe_gendered["count"]/total_tokens_uoe)*100
df_uoe_gendered.insert(len(df_uoe_gendered.columns), "percentage", perc_uoe)
df_uoe_gendered

count  percentage
word                          
Viscountess      0       0.000
Baroness         1       0.000
gentleman        1       0.000
gentlemen        1       0.000
Ms               2       0.000
lady             2       0.000
ladies           3       0.001
Messrs           4       0.001
Dame             5       0.001
Count            6       0.001
Countess         9       0.002
Viscount        10       0.002
Baron           15       0.003
Princess        21       0.004
Duchess         22       0.004
Prince          62       0.011
Duke            65       0.012
Miss            80       0.015
Mr              99       0.018
Queen          100       0.018
King           151       0.028
Lord           171       0.032
Lady           174       0.032
Mrs            184       0.034
Sir            406       0.075

In [36]:
df_uoe_gendered.describe()

count  percentage
count  25.000      25.000
mean   63.760       0.012
std    94.665       0.017
min     0.000       0.000
25%     3.000       0.001
50%    15.000       0.003
75%    99.000       0.018
max   406.000       0.075

There's greater variation in Newcastle's use of gendered terms in the second set while Edinburgh's has greater variation (double Newcastle's) for the first set.

## Parts of Speech

In [ ]:
descriptor_files = [8, 9] # for adj and adv, respectively

In [50]:
df_nusc_descriptor_stats = pd.read_csv(dir_nusc + files_nusc[descriptor_files[1]], index_col=0)
df_nusc_descriptor_stats

adj_by_desc  adv_by_desc
mean           4.515        1.340
median         1.000        0.000
minimum        0.000        0.000
maximum      101.000       45.000
total     127705.000    37915.000

In [57]:
print("Total adjectives as a percentage of all NUSC tokens:", str((df_nusc_descriptor_stats["adj_by_desc"]["total"]/total_tokens_nusc)*100)+"%")
print("Total adverbs as a percentage of all NUSC tokens:", str((df_nusc_descriptor_stats["adv_by_desc"]["total"]/total_tokens_nusc)*100)+"%")

Total adjectives as a percentage of all NUSC tokens: 6.66738019286091%
Total adverbs as a percentage of all NUSC tokens: 1.97951309668628%


In [51]:
df_uoe_descriptor_stats = pd.read_csv(dir_uoe + files_uoe[descriptor_files[1]], index_col=0)
df_uoe_descriptor_stats

adj_by_desc  adv_by_desc
mean           2.625        0.467
median         2.000        0.000
minimum        0.000        0.000
maximum     1290.000      252.000
total      32432.000     5763.000

In [58]:
print("Total adjectives as a percentage of all UoE tokens:", str((df_uoe_descriptor_stats["adj_by_desc"]["total"]/total_tokens_uoe)*100)+"%")
print("Total adverbs as a percentage of all UoE tokens:", str((df_uoe_descriptor_stats["adv_by_desc"]["total"]/total_tokens_uoe)*100)+"%")

Total adjectives as a percentage of all UoE tokens: 5.976761543210445%
Total adverbs as a percentage of all UoE tokens: 1.0620398610484028%


The average use of adjectives and adverbs in the Newcastle descriptions is roughly double that of Edinburgh, however the median use of adjectives in the Edinburgh dataset is double that of Newcastle (2 vs. 1).  Most noticeable are the differences in *maximum* number of adjectives or adverbs that appear in a description: Edinburgh's is far larger than Newcastles, with the maximum adjective count over 10x as large and the maximum adverb count over 5x as large.

As a percentage of all tokens per dataset, there's not a big difference, with Newcastle's data actually having a slightly larger percentage of descriptors relative to all tokens!

### Adjectives

In [59]:
adjective_files = [4, 5]  # the second is with lowercased tokens

In [81]:
df_adj_nusc = pd.read_csv(dir_nusc + files_nusc[adjective_files[1]], index_col=0)
df_adj_nusc.insert(1, "percentage_of_all_tokens", (df_adj_nusc["count"]/total_tokens_nusc)*100)
print(df_adj_nusc.shape)
df_adj_nusc.head(30)

(6122, 2)


count  percentage_of_all_tokens
token_lower                                   
mid-twentieth   5386                     0.281
key             3686                     0.192
historic        3650                     0.191
influential     3610                     0.188
notable         3099                     0.162
significant     2602                     0.136
academic        2356                     0.123
successful      2349                     0.123
polar           2312                     0.121
public          2144                     0.112
such            2010                     0.105
recent          2001                     0.104
partial         1967                     0.103
british         1943                     0.101
several         1933                     0.101
professional    1908                     0.100
major           1876                     0.098
distinctive     1806                     0.094
post-war        1805                     0.094
accessible      1802                     0.094
remarkable      1802                     0.094
"'new"          1801                     0.094
aesthetic       1798                     0.094
scholarly       1797                     0.094
all-time        1796                     0.094
new-wave        1795                     0.094
widely-read     1795                     0.094
polemical       1795                     0.094
many            1794                     0.094
other           1787                     0.093

In [106]:
df_adj_nusc.describe()

count  percentage_of_all_tokens
count 6122.000                  6122.000
mean    20.864                     0.001
std    178.966                     0.009
min      1.000                     0.000
25%      1.000                     0.000
50%      1.000                     0.000
75%      3.000                     0.000
max   5386.000                     0.281

In [82]:
df_adj_uoe = pd.read_csv(dir_uoe + files_uoe[adjective_files[1]], index_col=0)
df_adj_uoe.insert(1, "percentage_of_all_tokens", (df_adj_uoe["count"]/total_tokens_uoe)*100)
print(df_adj_uoe.shape)
df_adj_uoe.head(30)

(6889, 2)


count  percentage_of_all_tokens
token_lower                                 
20th          2143                     0.395
early         1569                     0.289
various        478                     0.088
late           468                     0.086
scottish       415                     0.076
other          384                     0.071
british        367                     0.068
first          307                     0.057
early/mid      264                     0.049
next           253                     0.047
domestic       220                     0.041
several        213                     0.039
old            206                     0.038
small          187                     0.034
new            174                     0.032
different      170                     0.031
young          169                     0.031
second         165                     0.030
many           160                     0.029
sheep          158                     0.029
indian         154                     0.028
genetic        152                     0.028
large          149                     0.027
general        122                     0.022
special        121                     0.022
original       118                     0.022
handwritten    114                     0.021
undated        112                     0.021
loose          107                     0.020
high           107                     0.020

In [105]:
df_adj_uoe.describe()

count  percentage_of_all_tokens
count 6889.000                  6889.000
mean     4.708                     0.001
std     36.273                     0.007
min      1.000                     0.000
25%      1.000                     0.000
50%      1.000                     0.000
75%      2.000                     0.000
max   2143.000                     0.395

Interesting to see that Newcastle's top adjectives include words that seem more likely to be communicate a judgment (e.g., "notable," "significant," "key," "influential") than Edinburgh's top adjectives!

In [95]:
common_adjs = df_adj_nusc.merge(df_adj_uoe, on="token_lower", how="inner", sort=True, suffixes=("_nusc", "_uoe"))
common_adjs.head()

count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
token_lower                                                         
1-10                  4                          0.000          3   
1-2                   7                          0.000          3   
1-22                  1                          0.000          2   
1-3                  14                          0.001          4   
1-4                  15                          0.001          3   

             percentage_of_all_tokens_uoe  
token_lower                                
1-10                                0.001  
1-2                                 0.001  
1-22                                0.000  
1-3                                 0.001  
1-4                                 0.001

Ignore numerical adjectives.

In [96]:
common_adjs = common_adjs.reset_index()
common_adjs_alpha = common_adjs.loc[common_adjs["token_lower"].str.isalpha()]
print(common_adjs.shape, common_adjs_alpha.shape)

(1651, 5) (1418, 5)


Look at the top adjectives for Newcastle...

In [107]:
common_adjs_alpha = common_adjs_alpha.sort_values(by="count_nusc", ascending=False)
common_adjs_alpha.head(30)

token_lower  count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
855            key        3686                          0.192         22   
757       historic        3650                          0.191          1   
812    influential        3610                          0.188          4   
1041       notable        3099                          0.162         10   
1364   significant        2602                          0.136          5   
171       academic        2356                          0.123         40   
1436    successful        2349                          0.123         25   
1154         polar        2312                          0.121          2   
1223        public        2144                          0.112         52   
1437          such        2010                          0.105         97   
1251        recent        2001                          0.104         22   
1113       partial        1967                          0.103         21   
308        british        1943                          0.101        367   
1346       several        1933                          0.101        213   
1199  professional        1908                          0.100         38   
937          major        1876                          0.098         32   
505    distinctive        1806                          0.094          2   
1273    remarkable        1802                          0.094          3   
194      aesthetic        1798                          0.094          2   
1319     scholarly        1797                          0.094          2   
944           many        1794                          0.094        160   
1090         other        1787                          0.093        384   
976       military        1628                          0.085         32   
640          first        1311                          0.068        307   
867          large        1200                          0.063        149   
530          early         976                          0.051       1569   
935           main         935                          0.049         40   
688        further         858                          0.045         51   
991           most         840                          0.044         47   
806     individual         819                          0.043         47   

      percentage_of_all_tokens_uoe  
855                          0.004  
757                          0.000  
812                          0.001  
1041                         0.002  
1364                         0.001  
171                          0.007  
1436                         0.005  
1154                         0.000  
1223                         0.010  
1437                         0.018  
1251                         0.004  
1113                         0.004  
308                          0.068  
1346                         0.039  
1199                         0.007  
937                          0.006  
505                          0.000  
1273                         0.001  
194                          0.000  
1319                         0.000  
944                          0.029  
1090                         0.071  
976                          0.006  
640                          0.057  
867                          0.027  
530                          0.289  
935                          0.007  
688                          0.009  
991                          0.009  
806                          0.009

...and for Edinburgh.

In [110]:
common_adjs_alpha = common_adjs_alpha.sort_values(by="count_uoe", ascending=False)
common_adjs_alpha.head(30)

token_lower  count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
530         early         976                          0.051       1569   
1589      various         777                          0.041        478   
872          late          61                          0.003        468   
1321     scottish          65                          0.003        415   
1090        other        1787                          0.093        384   
308       british        1943                          0.101        367   
640         first        1311                          0.068        307   
1028         next          82                          0.004        253   
513      domestic         106                          0.006        220   
1346      several        1933                          0.101        213   
1063          old          75                          0.004        206   
1382        small         414                          0.022        187   
1025          new         606                          0.032        174   
493     different         206                          0.011        170   
1646        young         122                          0.006        169   
1326       second         285                          0.015        165   
944          many        1794                          0.094        160   
1352        sheep           2                          0.000        158   
804        indian          67                          0.003        154   
699       genetic           5                          0.000        152   
867         large        1200                          0.063        149   
697       general         450                          0.023        122   
1406      special         184                          0.010        121   
1087     original         594                          0.031        118   
743   handwritten         125                          0.007        114   
1544      undated         111                          0.006        112   
753          high          66                          0.003        107   
926         loose          61                          0.003        107   
960       medical         254                          0.013        100   
205      american         104                          0.005        100   

      percentage_of_all_tokens_uoe  
530                          0.289  
1589                         0.088  
872                          0.086  
1321                         0.076  
1090                         0.071  
308                          0.068  
640                          0.057  
1028                         0.047  
513                          0.041  
1346                         0.039  
1063                         0.038  
1382                         0.034  
1025                         0.032  
493                          0.031  
1646                         0.031  
1326                         0.030  
944                          0.029  
1352                         0.029  
804                          0.028  
699                          0.028  
867                          0.027  
697                          0.022  
1406                         0.022  
1087                         0.022  
743                          0.021  
1544                         0.021  
753                          0.020  
926                          0.020  
960                          0.018  
205                          0.018

Export the DataFrame as a CSV file.

In [111]:
common_adjs_alpha.to_csv(comparison_dir+"common_adjs_sorted_nusc_uoe.csv")

Look at how many of Newcastle and Edinburgh's top 30 most common adjectives overlap.

In [142]:
top = 50

In [143]:
df_adj_nusc_top = df_adj_nusc.head(top)
df_adj_uoe_top = df_adj_uoe.head(top)
df_adj_top = df_adj_nusc_top.merge(df_adj_uoe_top, on="token_lower", how="outer", sort=True, suffixes=("_nusc", "_uoe")).reset_index()
df_adj_top = df_adj_top.sort_values(by="token_lower")
df_adj_top

token_lower  count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
0        "'new"    1801.000                          0.094        NaN   
1     1955-1958     770.000                          0.040        NaN   
2          19th         NaN                            NaN     76.000   
3          20th         NaN                            NaN   2143.000   
4      academic    2356.000                          0.123        NaN   
..          ...         ...                            ...        ...   
85      various     777.000                          0.041    478.000   
86        white         NaN                            NaN     79.000   
87         wide     790.000                          0.041        NaN   
88  widely-read    1795.000                          0.094        NaN   
89        young         NaN                            NaN    169.000   

    percentage_of_all_tokens_uoe  
0                            NaN  
1                            NaN  
2                          0.014  
3                          0.395  
4                            NaN  
..                           ...  
85                         0.088  
86                         0.015  
87                           NaN  
88                           NaN  
89                         0.031  

[90 rows x 5 columns]

In [145]:
df_adj_top_common = df_adj_top.loc[df_adj_top["count_nusc"].notna()]
df_adj_top_common = df_adj_top_common.loc[df_adj_top_common["count_uoe"].notna()]
print(f"Overlapping tokens in top {top} adjectives:", df_adj_top_common.shape[0])
df_adj_top_common

Overlapping tokens in top 50 adjectives: 10


token_lower  count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
12     british    1943.000                          0.101    367.000   
17       early     976.000                          0.051   1569.000   
20       first    1311.000                          0.068    307.000   
36       large    1200.000                          0.063    149.000   
42        many    1794.000                          0.094    160.000   
56       other    1787.000                          0.093    384.000   
71  scientific     816.000                          0.043     78.000   
74     several    1933.000                          0.101    213.000   
81        such    2010.000                          0.105     97.000   
85     various     777.000                          0.041    478.000   

    percentage_of_all_tokens_uoe  
12                         0.068  
17                         0.289  
20                         0.057  
36                         0.027  
42                         0.029  
56                         0.071  
71                         0.014  
74                         0.039  
81                         0.018  
85                         0.088

In [146]:
df_adj_top = df_adj_top.fillna("NA") # replace NaN with the string 'NA' (for "Not Applicable")

Save the DataFrame of the most common adjectives from both datasets.

In [147]:
df_adj_top.to_csv(comparison_dir+f"top{top}_adjs_nusc_uoe.csv")

### Adverbs

In [112]:
adverb_files = [6, 7]     # the second is with lowercased tokens

In [116]:
df_adv_nusc = pd.read_csv(dir_nusc + files_nusc[adverb_files[1]], index_col=0)
df_adv_nusc.insert(1, "percentage_of_all_tokens", (df_adv_nusc["count"]/total_tokens_nusc)*100)
print(df_adv_nusc.shape)
df_adv_nusc.head(30)

(622, 2)


count  percentage_of_all_tokens
token_lower                                  
also           6188                     0.323
most           2661                     0.139
more           1884                     0.098
particularly   1868                     0.098
primarily      1838                     0.096
ever           1810                     0.094
earlier        1804                     0.094
later          1801                     0.094
furthermore    1795                     0.094
hugely         1795                     0.094
markedly       1795                     0.094
letter         1706                     0.089
as             1303                     0.068
well           1280                     0.067
not             847                     0.044
largely         830                     0.043
too             797                     0.042
perhaps         779                     0.041
mainly          217                     0.011
mostly          213                     0.011
now             169                     0.009
then            164                     0.009
family          163                     0.009
originally      159                     0.008
together        158                     0.008
first           157                     0.008
very            139                     0.007
possibly        123                     0.006
only            115                     0.006
back             79                     0.004

In [117]:
df_adv_uoe = pd.read_csv(dir_uoe + files_uoe[adverb_files[1]], index_col=0)
df_adv_uoe.insert(1, "percentage_of_all_tokens", (df_adv_uoe["count"]/total_tokens_uoe)*100)
print(df_adv_uoe.shape)
df_adv_uoe.head(30)

(599, 2)


count  percentage_of_all_tokens
token_lower                                  
also            799                     0.147
not             425                     0.078
early           406                     0.075
then            306                     0.056
as              146                     0.027
well            143                     0.026
together        130                     0.024
later           123                     0.023
first           114                     0.021
possibly        112                     0.021
now              91                     0.017
only             88                     0.016
mainly           73                     0.013
probably         66                     0.012
very             65                     0.012
still            65                     0.012
there            57                     0.011
too              53                     0.010
most             50                     0.009
so               48                     0.009
however          48                     0.009
more             46                     0.008
presumably       46                     0.008
again            42                     0.008
particularly     42                     0.008
largely          40                     0.007
back             40                     0.007
mostly           37                     0.007
just             37                     0.007
here             35                     0.006

There's lots of overlap in Edinburgh and Newcastle's top adverbs.  They seem to be used to communicate a time estimate or to communicate an uncertainty or likelihood.

In [118]:
common_advs = df_adv_nusc.merge(df_adv_uoe, on="token_lower", how="inner", sort=True, suffixes=("_nusc", "_uoe"))
common_advs.head()

count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
token_lower                                                         
about                 9                          0.000          6   
abroad               26                          0.001          6   
absolutely            1                          0.000          1   
accordingly           1                          0.000          3   
actively              9                          0.000          2   

             percentage_of_all_tokens_uoe  
token_lower                                
about                               0.001  
abroad                              0.001  
absolutely                          0.000  
accordingly                         0.001  
actively                            0.000

Look at the common adverbs by Newcastle's most common...

In [119]:
common_advs = common_advs.sort_values(by="count_nusc", ascending=False)
common_advs.head(30)

count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
token_lower                                                          
also                6188                          0.323        799   
most                2661                          0.139         50   
more                1884                          0.098         46   
particularly        1868                          0.098         42   
primarily           1838                          0.096         12   
ever                1810                          0.094         14   
earlier             1804                          0.094         24   
later               1801                          0.094        123   
furthermore         1795                          0.094          1   
hugely              1795                          0.094          1   
letter              1706                          0.089          1   
as                  1303                          0.068        146   
well                1280                          0.067        143   
not                  847                          0.044        425   
largely              830                          0.043         40   
too                  797                          0.042         53   
perhaps              779                          0.041         26   
mainly               217                          0.011         73   
mostly               213                          0.011         37   
now                  169                          0.009         91   
then                 164                          0.009        306   
family               163                          0.009         11   
originally           159                          0.008         31   
together             158                          0.008        130   
first                157                          0.008        114   
very                 139                          0.007         65   
possibly             123                          0.006        112   
only                 115                          0.006         88   
back                  79                          0.004         40   
yet                   77                          0.004         16   

              percentage_of_all_tokens_uoe  
token_lower                                 
also                                 0.147  
most                                 0.009  
more                                 0.008  
particularly                         0.008  
primarily                            0.002  
ever                                 0.003  
earlier                              0.004  
later                                0.023  
furthermore                          0.000  
hugely                               0.000  
letter                               0.000  
as                                   0.027  
well                                 0.026  
not                                  0.078  
largely                              0.007  
too                                  0.010  
perhaps                              0.005  
mainly                               0.013  
mostly                               0.007  
now                                  0.017  
then                                 0.056  
family                               0.002  
originally                           0.006  
together                             0.024  
first                                0.021  
very                                 0.012  
possibly                             0.021  
only                                 0.016  
back                                 0.007  
yet                                  0.003

...and by Edinburgh's most common.

In [120]:
common_advs = common_advs.sort_values(by="count_uoe", ascending=False)
common_advs.head(30)

count_nusc  percentage_of_all_tokens_nusc  count_uoe  \
token_lower                                                          
also                6188                          0.323        799   
not                  847                          0.044        425   
early                 30                          0.002        406   
then                 164                          0.009        306   
as                  1303                          0.068        146   
well                1280                          0.067        143   
together             158                          0.008        130   
later               1801                          0.094        123   
first                157                          0.008        114   
possibly             123                          0.006        112   
now                  169                          0.009         91   
only                 115                          0.006         88   
mainly               217                          0.011         73   
probably              26                          0.001         66   
still                 56                          0.003         65   
very                 139                          0.007         65   
there                 18                          0.001         57   
too                  797                          0.042         53   
most                2661                          0.139         50   
so                    73                          0.004         48   
however               65                          0.003         48   
more                1884                          0.098         46   
presumably            49                          0.003         46   
again                 32                          0.002         42   
particularly        1868                          0.098         42   
back                  79                          0.004         40   
largely              830                          0.043         40   
just                  36                          0.002         37   
mostly               213                          0.011         37   
here                  75                          0.004         35   

              percentage_of_all_tokens_uoe  
token_lower                                 
also                                 0.147  
not                                  0.078  
early                                0.075  
then                                 0.056  
as                                   0.027  
well                                 0.026  
together                             0.024  
later                                0.023  
first                                0.021  
possibly                             0.021  
now                                  0.017  
only                                 0.016  
mainly                               0.013  
probably                             0.012  
still                                0.012  
very                                 0.012  
there                                0.011  
too                                  0.010  
most                                 0.009  
so                                   0.009  
however                              0.009  
more                                 0.008  
presumably                           0.008  
again                                0.008  
particularly                         0.008  
back                                 0.007  
largely                              0.007  
just                                 0.007  
mostly                               0.007  
here                                 0.006

## COME BACK TO: run analysis and comparison code on sample NUSC data that was manually reviewed!